### DISCUSSION TITLE
block-diagram generation tools

### DISCUSSION THREAD

**Discussion by biosbob** [stnolting/neorv32#624] [Q&A]
i would like to create a (hierarchical) block-diagram of the  RTL, starting from `neorv32_top`....  i've read about `netlistsvg`, but wanted to ask if there were any other preferred options???

also, is it possible to create an "interactive" block-diagram in an environment like vivado -- enabling me to zoom in/out at will on sub-blocks????

suggestion:  could something be generated as part of the CI flow????

**Comment by stnolting** [comment]
I have no idea about any open-source tools capable of generating this (but I would be really interested in that!), but I am also not quite sure what level of details you need.

Maybe TEROS HDL might be worth a try: https://github.com/stnolting/neorv32/issues/94
